components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal

```diff
@@ -395,7 +393,7 @@ public function merge(json target, json source) returns (json) {
         } else {
             //if there is a value available for this property in the source model, check whether there are any
             //exceptional keys inside that property
-            if (sourceProperty.getKeys() != null) {
+            if (sourceProperty.getKeys().length() > 0) {
```

Comment:
What if  sourceProperty.getKeys() is null, then calling the length() of a null will be a issue I guess